src/parsers/parsePackageJson.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { pathExists, readJson } from 'fs-extra';
+import * as path from 'path';
+import { ensureErrorType } from '../utils/ensureErrorType';
+
+export interface PackageJson {
+    type?: string;
+    main?: string;
+}
+
+/**
+ * @returns A parsed & sanitized package.json
+ */
+export async function parsePackageJson(dir: string): Promise<PackageJson> {
+    try {
+        const filePath = path.join(dir, 'package.json');
+        if (!(await pathExists(filePath))) {
+            throw new Error('file does not exist');
+        }
+
+        const data: unknown = await readJson(filePath);
+        if (typeof data !== 'object' || data === null || Array.isArray(data)) {
+            throw new Error('file content is not an object');
+        }
+
+        const stringFields = ['main', 'type'];
+        for (const field of stringFields) {
+            if (field in data && typeof data[field] !== 'string') {
+                // ignore fields with an unexpected type
+                delete data[field];
+            }
+        }
```

Comment:
I'm open to suggestions, but I'm guessing it'll be more complicated than it's worth. Mostly because I don't think we will use very many more properties from package.json at all, but slightly because I would hope someone adding a property to the *parse*PackageJson file would know they might need to add logic to _parse_ the property

Reply:
I think you're right. I was hoping there was an easy way to do this but apparently not :p